### PR TITLE
Bsw 2496

### PR DESCRIPTION
**Pull request by lilChemsHulkHogan**

### CODE REVIEW

File: src/widgets/WalletModal/ConnectModal.tsx [hunk 1]
```diff
@@ -88,16 +133,33 @@ const ConnectModal: React.FC<Props> = ({ login, onDismiss = () => null }) => {
...
+      </ScrollWrapper>
+      <Text mt="24px" textAlign="center" fontSize="12px">
+        Haven’t got a crypto wallet yet?
+      </Text>
+      <Button
+        height="48px"
+        width={isMobile ? "306px" : "352px"}
+        as="a"
+        color="primary"
+        m={isMobile ? "16px 16px 16px" : "16px 32px 32px"}
```
Comment (front001) [inline src/widgets/WalletModal/ConnectModal.tsx:145]: попробуй коротший опис

File: src/widgets/WalletModal/ConnectModal.tsx [hunk 2]
```diff
@@ -88,16 +133,33 @@ const ConnectModal: React.FC<Props> = ({ login, onDismiss = () => null }) => {
...
+        href="https://docs.biswap.org/faq/biswap-platform#how-do-i-connect-my-wallet-to-biswap"
```
Comment (front001) [inline src/widgets/WalletModal/ConnectModal.tsx:146]: люблю константи пезда

File: src/widgets/WalletModal/ConnectModal.tsx [hunk 3]
```diff
@@ -76,8 +102,27 @@ const ConnectModal: React.FC<Props> = ({ login, onDismiss = () => null }) => {
...
-      <Flex flexDirection="column">
+    <Modal
+      walletModal
+      title="Connect to a wallet"
+      onDismiss={onDismiss}
+      width={isMobile ? "100%" : "auto"}
+      maxWidth={!isMobile ? "416px" : "none"}
+      bodyPadding="0"
+      position={isMobile ? "absolute" : "relative"}
+      bottom={isMobile ? "0" : null}
```
Comment (front001) [inline src/widgets/WalletModal/ConnectModal.tsx:113]: глянь до цих пропсів

File: src/widgets/WalletModal/ConnectModal.tsx [hunk 4]
```diff
@@ -59,6 +57,34 @@ const getPreferredConfig = (walletConfig: Config[]) => {
...
+  &::-webkit-scrollbar-thumb {
+    background-color: ${({ theme }) => theme.colors.textSubtle};
+  }
+`;
+
+const StyledText = styled(Text)`
+  align-self: flex-start;
+`;
+
+const StyledButton = styled(Button)`
```
Comment (front001) [inline src/widgets/WalletModal/ConnectModal.tsx:83]: точно треба шоб був 500 font weight? а то фігма часто бажить, і в нас нема кнопки з 16px?

Comment (andkfront) [reply src/widgets/WalletModal/ConnectModal.tsx:83]: ++ в нас лише два варіка жирності. 400 і 600

File: src/components/DropdownMenu/MobileMenu/MobileMenu.tsx [hunk 5]
```diff
@@ -29,8 +29,7 @@ const MobileCommunityWrapper = styled.div`
 const StyledMobileMenu = styled.div<{
   $isOpen: boolean;
 }>`
-  margin-top: 48px;
-  padding: 0 24px 0 24px;
+  padding: 32px 24px 0 24px;
```
Comment (chefReyMysterio) [inline src/components/DropdownMenu/MobileMenu/MobileMenu.tsx]: padding: 32px 24px 0; ????

File: src/components/DropdownMenu/types.ts [hunk 6]
```diff
@@ -8,6 +8,7 @@ export interface MobileMenuProps {
   mobileMenuCallback?: (s: boolean) => void;
   activeItem?: string;
   isMobileNav?: boolean;
+  children: ReactNode;
```
Comment (andkfront) [inline src/components/DropdownMenu/types.ts:11]: точно обовʼязкова має бути?

File: src/widgets/Menu/components/UserMenu/types.ts [hunk 7]
```diff
@@ -14,6 +15,7 @@ export interface UserMenuProps extends FlexProps {
   text?: string;
   avatarSrc?: string;
   variant?: Variant;
+  children: ReactNode;
```
Comment (andkfront) [inline src/widgets/Menu/components/UserMenu/types.ts:18]: і тут? не буде меню орать?

File: src/widgets/WalletModal/ConnectModal.tsx [hunk 8]
```diff
@@ -1,36 +1,34 @@
...
 const WalletCardsWrapper = styled.div`
   display: grid;
-  grid-gap: 16px;
-  width: 100%;
+  grid-gap: 8px;
   grid-template-columns: repeat(2, 1fr);
+  max-height: 328px;
+  margin-left: 16px;
+  padding-right: 6px;
+  ${({ theme }) => theme.mediaQueries.sm} {
```
Comment (ghost) [inline src/widgets/WalletModal/ConnectModal.tsx:32]: yarn format

File: src/widgets/WalletModal/ConnectModal.tsx [hunk 9]
```diff
@@ -59,6 +57,34 @@ const getPreferredConfig = (walletConfig: Config[]) => {
   ];
 };
 
+const ScrollWrapper = styled(Box)`
+  overflow-x: hidden;
+  margin-right: 6px;
+  align-self: stretch;
+  ${({ theme }) => theme.mediaQueries.sm} {
```
Comment (ghost) [inline src/widgets/WalletModal/ConnectModal.tsx:70]: yarn format

File: src/widgets/WalletModal/ConnectModal.tsx [hunk 10]
```diff
@@ -1,36 +1,34 @@
...
-  align-self: center;
-  align-items: center;
-  margin-top: 40px;
-
-  ${({ theme }) => theme.mediaQueries.sm} {
-    margin-top: 24px;
-  }
-`;
-
 const WalletCardsWrapper = styled.div`
```
Comment (andkfront) [inline src/widgets/WalletModal/ConnectModal.tsx:20]: в нас є компонент Grid

File: src/widgets/WalletModal/WalletCard.tsx [hunk 11]
```diff
@@ -26,23 +30,31 @@ const StyledText = styled(Text)`
...
-        window.localStorage.setItem(
-          connectorLocalStorageKey,
-          walletConfig.connectorId
-        );
-        onDismiss();
-      }}
+      onClick={handleClick}
       id={`wallet-connect-${title.toLocaleLowerCase()}`}
     >
       <Icon width="32px" />
```
Comment (andkfront) [inline src/widgets/WalletModal/WalletCard.tsx:54]: не вийде Icon предати пропсою startIcon в кнопку???

File: src/components/DropdownMenu/MobileMenu/MobileMenu.tsx (hunk 5 above)
Comment (chefReyMysterio) [inline src/components/DropdownMenu/MobileMenu/MobileMenu.tsx]: padding: 32px 24px 0; ??

File: src/components/DropdownMenu/types.ts (hunk 6 above)
Comment (chefReyMysterio) [inline src/components/DropdownMenu/types.ts]: this props is necessary required??

File: src/widgets/WalletModal/ConnectModal.tsx [hunk 12]
```diff
@@ -59,6 +57,34 @@ const getPreferredConfig = (walletConfig: Config[]) => {
...
+  }
+  &::-webkit-scrollbar {
+    width: 4px;
+  }
+  &::-webkit-scrollbar-track {
+    background-color: ${({ theme }) => theme.colors.gray200};
+    box-shadow: none;
+  }
+  &::-webkit-scrollbar-thumb {
+    background-color: ${({ theme }) => theme.colors.textSubtle};
```
Comment (ghost) [inline src/widgets/WalletModal/ConnectModal.tsx]: this is a old color, use new

File: src/widgets/WalletModal/ConnectModal.tsx (hunk 1 above)
Comment (chefReyMysterio) [inline src/widgets/WalletModal/ConnectModal.tsx]: replace "16px 16px 16px" to "16px"

File: src/widgets/WalletModal/ConnectModal.tsx [hunk 13]
```diff
@@ -59,6 +57,34 @@ const getPreferredConfig = (walletConfig: Config[]) => {
...
+  }
+`;
+
+const StyledText = styled(Text)`
+  align-self: flex-start;
+`;
+
+const StyledButton = styled(Button)`
+  font-size: 16px;
+  font-weight: 500;
```
Comment (ghost) [inline src/widgets/WalletModal/ConnectModal.tsx]: styled system don`t have font-weight: 500
only 400 & 600
if you need bold text, use 600

File: src/widgets/WalletModal/WalletCard.tsx [hunk 14]
```diff
@@ -13,11 +13,15 @@ interface Props {
...
 const StyledButton = styled(Button)`
   display: flex;
-  flex-direction: column;
-  justify-content: space-around;
-  padding: 8px 0 0;
+  flex-direction: row;
+  justify-content: flex-start;
   white-space: nowrap;
-  min-height: 91px;
+  height: 48px;
```
Comment (chefReyMysterio) [inline src/widgets/WalletModal/WalletCard.tsx]: XL scale has height 48px

File: src/widgets/WalletModal/ConnectModal.tsx (hunk 4 above)
Comment (ghost) [inline src/widgets/WalletModal/ConnectModal.tsx]: why styled button, styled system and ui-kit have more complete buttons

File: src/widgets/WalletModal/ConnectModal.tsx [hunk 15]
```diff
@@ -88,16 +133,33 @@ const ConnectModal: React.FC<Props> = ({ login, onDismiss = () => null }) => {
             />
           ))}
         </WalletCardsWrapper>
-        <HelpLink
-          href="https://docs.biswap.org/faq/biswap-platform#how-do-i-connect-my-wallet-to-biswap"
-          external
+      </ScrollWrapper>
+      <Text mt="24px" textAlign="center" fontSize="12px">
```
Comment (ghost) [inline src/widgets/WalletModal/ConnectModal.tsx]: add tag please, as="span" or "p" and more

File: src/widgets/WalletModal/WalletCard.tsx (hunk 11 above)
Comment (chefReyMysterio) [inline src/widgets/WalletModal/WalletCard.tsx]: check please. can the Icon be passed as a startIcon props to Button

File: src/widgets/WalletModal/ConnectModal.tsx [hunk 16]
```diff
@@ -88,16 +133,33 @@ const ConnectModal: React.FC<Props> = ({ login, onDismiss = () => null }) => {
...
+      <Button
+        height="48px"
+        width={isMobile ? "306px" : "352px"}
+        as="a"
+        color="primary"
+        m={isMobile ? "16px 16px 16px" : "16px 32px 32px"}
+        href="https://docs.biswap.org/faq/biswap-platform#how-do-i-connect-my-wallet-to-biswap"
+      >
+        <HelpOpacityIcon color="white" />
+        <Text color="contrast" bold ml="12px">
```
Comment (ghost) [inline src/widgets/WalletModal/ConnectModal.tsx]: add tag

File: src/widgets/WalletModal/WalletCard.tsx [hunk 17]
```diff
@@ -13,11 +13,15 @@ interface Props {
 
 const StyledButton = styled(Button)`
   display: flex;
-  flex-direction: column;
-  justify-content: space-around;
-  padding: 8px 0 0;
+  flex-direction: row;
```
Comment (ghost) [inline src/widgets/WalletModal/WalletCard.tsx]: "row" is a default property

File: src/widgets/WalletModal/WalletCard.tsx [hunk 18]
```diff
@@ -13,11 +13,15 @@ interface Props {
...
+  justify-content: flex-start;
   white-space: nowrap;
-  min-height: 91px;
+  height: 48px;
+  width: 100%;
+  background: ${({ theme }) => theme.colors.gray200};
```
Comment (ghost) [inline src/widgets/WalletModal/WalletCard.tsx:20]: why don't you put an empty line in such cases, it's not readable

File: src/widgets/WalletModal/ConnectModal.tsx (hunk 3 above)
Comment (chefalexbs) [inline src/widgets/WalletModal/ConnectModal.tsx]: check this props

File: src/widgets/WalletModal/ConnectModal.tsx (hunk 2 above)
Comment (chefalexbs) [inline src/widgets/WalletModal/ConnectModal.tsx]: in const

File: src/widgets/WalletModal/WalletCard.tsx [hunk 19]
```diff
@@ -26,23 +30,31 @@ const StyledText = styled(Text)`
 `;
 
 const WalletCard: React.FC<Props> = ({ login, walletConfig, onDismiss }) => {
+  const handleClick = useCallback(() => {
```
Comment (chefalexbs) [inline src/widgets/WalletModal/WalletCard.tsx]: useCallback haven't deps, useCallback should be used here?

File: src/widgets/WalletModal/ConnectModal.tsx (hunk 12 above)
Comment (lilChemsHulkHogan) [reply src/widgets/WalletModal/ConnectModal.tsx]: I don't know that type of color. Be more specific

File: src/widgets/WalletModal/WalletCard.tsx [hunk 20]
```diff
@@ -13,11 +13,15 @@ interface Props {
...
+  width: 100%;
+
+  background: ${({ theme }) => theme.colors.gray200};
+
+  ${({ theme }) => theme.mediaQueries.sm} {
+    width: 172px;
+  }
 `;
 
 const StyledText = styled(Text)`
```
Comment (ghost) [inline src/widgets/WalletModal/WalletCard.tsx]: i thick this styled text may delete, Text component have "bold" and "color" props

File: src/widgets/WalletModal/WalletCard.tsx [hunk 21]
```diff
@@ -26,25 +30,35 @@ const StyledText = styled(Text)`
...
-        onDismiss();
-      }}
+      onClick={handleClick}
       id={`wallet-connect-${title.toLocaleLowerCase()}`}
+      p="0 12px"
+      startIcon={<Icon width="32px" />}
     >
-      <Icon width="32px" />
-      <StyledText>{title}</StyledText>
+      <StyledText pl="8px" fontSize="12px">
```
Comment (ghost) [inline src/widgets/WalletModal/WalletCard.tsx]: may delete

**Pull request closed without merging**